ntirpc/rpc/gss_internal.h

```diff
@@ -113,30 +112,17 @@ svc_rpc_gss_data *alloc_svc_rpc_gss_data(void)
 }
 
 static inline void
-unref_svc_rpc_gss_data(struct svc_rpc_gss_data *gd,
-		       uint32_t flags)
+unref_svc_rpc_gss_data(struct svc_rpc_gss_data *gd)
 {
-	u_int refcnt;
-	bool gd_locked = flags & SVC_RPC_GSS_FLAG_LOCKED;
-
-	refcnt = atomic_dec_uint32_t(&gd->refcnt);
+	mutex_lock(&gd->lock);
```

Comment:
Why do you need the above lock? Some tools may complain about missing unlock as well with the current code. I imagine the lock was needed before the atomic ops.